selftests/functional/test_skiptests.py

```diff
@@ -80,9 +114,40 @@ def test_skip_decorators(self):
                     '--json -']
         result = process.run(' '.join(cmd_line), ignore_status=True)
         json_results = json.loads(result.stdout)
+        debuglog = json_results['debuglog']
+
         self.assertEqual(result.exit_status, exit_codes.AVOCADO_ALL_OK)
-        self.assertEquals(json_results['pass'], 2)
-        self.assertEquals(json_results['skip'], 4)
+        self.assertEquals(json_results['skip'], 3)
+        self.assertFalse('setup executed' in open(debuglog, 'r').read())
+        self.assertFalse('teardown executed' in open(debuglog, 'r').read())
```

Comment:
Btw while on it how about also adding check that the `test` method is not executed? (I know right now it does not make sense, but in the future we might introduce different layout and it might actually happen...)